api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/plugins/ContrastPluginServiceImpl.java

```diff
@@ -0,0 +1,175 @@
+        }
+        if (StringUtils.isEmpty(pluginData.getServiceKey())) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, "serviceKey"));
+        }
+        if (StringUtils.isEmpty(pluginData.getApiKey())) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, "apiKey"));
+        }
+        if (StringUtils.isEmpty(pluginData.getUserId())) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, "userId"));
+        }
+        String validationError = validationErrorDetails.toString();
+        if (!validationError.isEmpty()) {
+            response.setStatus(AdminConstants.FAILURE);
+            response.setMessage(String.format(INVALID_PLUGIN_REQUEST_MSG, pluginName));
+            response.setErrorDetails(validationError);
+        } else {
+            response.setStatus(AdminConstants.SUCCESS);
+        }
+        return response;
+    }
+
+    public String getOrganizationNameFromApiResponse(ContrastPluginRequest accountData) {
+        try {
+            String apiUrl = new URIBuilder(String.format(CONTRAST_URL_TEMPLATE, accountData.getEnvironmentName()))
+                    .setPath(ORGANIZATIONS_URL_PATH + accountData.getOrganizationId()).build().toString();
+            String authKey = Base64.getEncoder().encodeToString((accountData.getUserId() + ":"
+                    + accountData.getServiceKey()).getBytes());
+            HttpGet request = new HttpGet(apiUrl);
+            request.setHeader(HttpHeaders.AUTHORIZATION, authKey);
+            request.setHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+            request.setHeader("API-Key", accountData.getApiKey());
+            JsonNode jsonResponse = getApiResponse(request);
+            return jsonResponse.get("name").textValue();
+        } catch (PluginApiResponseException pare) {
+            String message;
+            switch (pare.getStatusCode()) {
+                case 400:
+                    message = "The request could not be understood by the server. " +
+                            "Check the Organization ID and Keys and make sure they are correct.";
+                    break;
+                case 401:
+                    message = "The request requires authentication. " +
+                            "The User Id and Keys may be incorrect or the User is locked.";
+                    break;
+                case 1000:
+                    message = "The server could not be found. " +
+                            "The Environment name may be incorrect or the server may be down.";
+                    break;
+                default:
+                    message = "The server does not have the necessary credentials to authenticate the user. " +
+                            "Check your credentials and make sure they are correct.";
+                    break;
+            }
+            throw new PluginApiResponseException(message);
```

Comment:
instead of throwing exceptions (they are heavy objects!) make this method to return status of response. Like any http client/server does.

Reply:
updated